wagtail/admin/views/generic/mixins.py

```diff
@@ -256,18 +256,36 @@ def setup_workflow(self):
         self.current_workflow_task = self.object.current_workflow_task
 
     def user_has_permission(self, permission):
-        # Check with base PermissionCheckedMixin logic before any complex checks
-        has_base_permission = super().user_has_permission(permission)
-        if has_base_permission:
+        user = self.request.user
+        if user.is_superuser:
             return True
 
-        # Allow unlocking even if the user does not have the 'unlock' permission
-        # if they are the user who locked the object
-        if permission == "unlock" and self.object.locked_by_id == self.request.user.pk:
+        # Workflow lock/unlock methods take precedence before the base
+        # "lock" and "unlock" permissions -- see PagePermissionTester for reference
```

Comment:
Not a big deal for this PR necessarily, but I wonder if moving towards a view-level permission checking here, vs a tester class in Page is causing the permissions system to diverge further. I think it would be nice to consolidate how we do this. (eg a class supplied by the permission policy?

Reply:
(more a general point for the future - I really feel like we should consolidate permissions checks)

Reply:
I agree, I had the same concern and yep I think this needs its own dedicated time for API design and implementation.